src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateEnvPlaceholdersPassTest.php

```diff
@@ -22,31 +22,31 @@
 
 class ValidateEnvPlaceholdersPassTest extends TestCase
 {
-    /**
-     * @expectedException \Symfony\Component\DependencyInjection\Exception\LogicException
-     * @expectedExceptionMessage Invalid type for env parameter "env(FOO)". Expected "string", but got "bool".
-     */
-    public function testDefaultEnvIsValidatedByType()
+    public function testEnvsAreValidatedInConfig()
     {
         $container = new ContainerBuilder();
-        $container->setParameter('env(FOO)', true);
-        $container->registerExtension(new EnvExtension());
-        $container->prependExtensionConfig('env_extension', array(
-            'scalar_node' => '%env(FOO)%',
+        $container->setParameter('env(NULLED)', null);
+        $container->setParameter('env(FLOATISH)', 3.2);
+        $container->registerExtension($ext = new EnvExtension());
+        $container->prependExtensionConfig('env_extension', $expected = array(
+            'scalar_node' => '%env(NULLED)%',
+            'scalar_node_not_empty' => '%env(FLOATISH)%',
+            'int_node' => '%env(int:FOO)%',
+            'float_node' => '%env(float:BAR)%',
         ));
 
         $this->doProcess($container);
+
+        $this->assertSame($expected, $container->resolveEnvPlaceholders($ext->getConfig()));
     }
 
-    public function testEnvsAreValidatedInConfig()
+    public function testDefaultEnvWithoutPrefixIsValidatedInConfig()
     {
         $container = new ContainerBuilder();
-        $container->setParameter('env(NULLED)', null);
+        $container->setParameter('env(FLOATISH)', 3.2);
         $container->registerExtension($ext = new EnvExtension());
         $container->prependExtensionConfig('env_extension', $expected = array(
-            'scalar_node' => '%env(NULLED)%',
-            'int_node' => '%env(int:FOO)%',
-            'float_node' => '%env(float:BAR)%',
+            'float_node' => '%env(FLOATISH)%',
```

Comment:
currently this fails, im not sure what we expect here. Env without a prefix is now assumed to be a scalar which is not allowed for float-only node.

Reply:
shouldn't we use `is_numeric` somewhere?

Reply:
It's a scalar issue in general, directly related to

https://github.com/symfony/symfony/blob/c81f88f38bc883fc8b8950f8840bef22db29eda0/src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php#L52-L54

im not sure what's the behavior we're seeking... what does "env without prefix" mean on its value? It if can be any scalar then using it for a float node is invalid, technically.

to make it work we should simply avoid the type check in config for this case, but it makes validation less strict.

Reply:
Would it be possible to prefix a processor here? Referencing the envelope bar with float: prepended?

Reply:
Well, the thing is, without the `float` cast in the env var retrieval, the service would not receive a float, but a string. So `float_node` not accepting `%env(FLOATISH)%` is expected to me.